x/stakeibc/keeper/msg_server_submit_tx.go

```diff
@@ -94,73 +94,32 @@ func (k Keeper) DelegateOnHost(ctx sdk.Context, hostZone types.HostZone, amt sdk
-		strideAmount := balanceDec.Mul(commission)
-		reinvestAmount := balanceDec.Sub(strideAmount)
-		strideCoin := sdk.NewCoin(hostZone.HostDenom, strideAmount.TruncateInt())
-		reinvestCoin := sdk.NewCoin(hostZone.HostDenom, reinvestAmount.TruncateInt())
-
-		// transfer balances from the withdraw address to the delegation account
-		sendBalanceToDelegationAccount := &bankTypes.MsgSend{FromAddress: withdrawAccount.GetAddress(), ToAddress: delegationAccount.GetAddress(), Amount: sdk.NewCoins(reinvestCoin)}
-		msgs = append(msgs, sendBalanceToDelegationAccount)
-		// TODO: [TEST-115] get the stride commission addresses (potentially split this up into multiple messages)
-		strideCommmissionAccount := "cosmos12vfkpj7lpqg0n4j68rr5kyffc6wu55dzqewda4"
-		sendBalanceToStrideAccount := &bankTypes.MsgSend{FromAddress: withdrawAccount.GetAddress(), ToAddress: strideCommmissionAccount, Amount: sdk.NewCoins(strideCoin)}
-		msgs = append(msgs, sendBalanceToStrideAccount)
-
-		// Send the transaction through SubmitTx
-		// TODO(TEST-5): Add a record with STATUS PENDING
-		// TODO(TEST-5): In the ack, update the record to STATUS STAKE
-		err = k.SubmitTxs(ctx, connectionId, msgs, *withdrawAccount)
-		if err != nil {
-			return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "Failed to SubmitTxs for %s, %s, %s", connectionId, hostZone.ChainId, msgs)
-		}
-
-		ctx.EventManager().EmitEvents(sdk.Events{
-			sdk.NewEvent(
-				sdk.EventTypeMessage,
-				sdk.NewAttribute("WithdrawalAccountBalance", balance.String()),
-				sdk.NewAttribute("ReinvestedPortion", reinvestCoin.String()),
-				sdk.NewAttribute("StrideCommission", strideCoin.String()),
-			),
-		})
-
-		return nil
-	}
-	// 1. query withdraw account balances using icq
-	// 2. transfer withdraw account balances to the delegation account in the cb
-	// 3. TODO: in the ICA ack upon transfer, reinvest those rewards and withdraw rewards
-	k.InterchainQueryKeeper.QueryBalances(ctx, hostZone, cb, withdrawAccount.Address)
-	return nil
+// Simple balance query helper using new ICQ module
+func (k Keeper) UpdateWithdrawalBalance(ctx sdk.Context, zoneInfo types.HostZone) {
+	k.Logger(ctx).Info(fmt.Sprintf("\tUpdating withdrawal balances on %s", zoneInfo.ChainId))
+
+	withdrawalIca := zoneInfo.GetWithdrawalAccount()
+	if withdrawalIca == nil || withdrawalIca.Address == "" {
+		k.Logger(ctx).Error("Zone %s is missing a delegation address!", zoneInfo.ChainId)
+	}
+	k.Logger(ctx).Info(fmt.Sprintf("\tQuerying withdrawalBalances for %s at %d height", zoneInfo.ChainId))
+
+	_, addr, _ := bech32.DecodeAndConvert(withdrawalIca.GetAddress())
+	data := bankTypes.CreateAccountBalancesPrefix(addr)
+	key := "store/bank/key"
+	k.Logger(ctx).Info("Querying for value", "key", key, "denom", zoneInfo.HostDenom) // debug?
+	k.InterchainQueryKeeper.MakeRequest(
+		ctx,
+		zoneInfo.ConnectionId,
+		zoneInfo.ChainId,
+		key,
+		append(data, []byte(zoneInfo.HostDenom)...),
+		sdk.NewInt(-1),
+		types.ModuleName,
+		"withdrawalbalance",
+		0, //ttl
```

Comment:
why is ttl 0?

Reply:
I believe nonzero ttl stores queries for more than one epoch rather then processing immediately. We don't need that feature